drivers/crypto/crypto_tc_shim.c

```diff
@@ -114,24 +114,13 @@ static int do_ccm_encrypt_mac(struct cipher_ctx *ctx,
 	}
 
 	if (tc_ccm_generation_encryption(op->out_buf, op->out_buf_max,
-					 aead_op->ad, aead_op->ad_len, op->in_buf,
-					 op->in_len, &ccm) == TC_CRYPTO_FAIL) {
+				aead_op->ad, aead_op->ad_len, op->in_buf,
+				op->in_len, &ccm, aead_op->tag) == TC_CRYPTO_FAIL) {
 		LOG_ERR("TC internal error during CCM Encryption OP");
 		return -EIO;
 	}
 
-	/* Looks like TinyCrypt appends the MAC to the end of out_buf as it
-	 * does not give a separate hash parameter. The user needs to be aware
-	 * of this and provide sufficient buffer space in output buffer to hold
-	 * both encrypted output and hash
-	 */
-	aead_op->tag = op->out_buf + op->in_len;
```

Comment:
instead of changing tinycrypt api, you can just:
`memcpy(aead_op->tag, op->out_buf + op->in_len, ccm.mlen)`

Reply:
this will ensure the behaviour described in our doc and do the same thing mbdetls does.

Reply:
my concern is that if tag behaved as doc described, users may just make sure the length of out_buf is op->in_len.